homeassistant/components/steam_online/config_flow.py

```diff
@@ -0,0 +1,211 @@
+"""Config flow for Steam integration."""
+from __future__ import annotations
+
+from typing import Any
+
+import steam
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_API_KEY, Platform
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import config_validation as cv, entity_registry as er
+from homeassistant.helpers.typing import ConfigType
+
+from .const import CONF_ACCOUNT, CONF_ACCOUNTS, DEFAULT_NAME, DOMAIN, LOGGER
+
+
+def validate_input(user_input: dict[str, str | int]) -> list[dict[str, str | int]]:
+    """Handle common flow input validation."""
+    steam.api.key.set(user_input[CONF_API_KEY])
+    interface = steam.api.interface("ISteamUser")
+    names = interface.GetPlayerSummaries(steamids=user_input[CONF_ACCOUNT])
+    return names["response"]["players"]["player"]
+
+
+class SteamFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Steam."""
+
+    def __init__(self) -> None:
+        """Initialize the flow."""
+        self.entry: config_entries.ConfigEntry | None = None
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: config_entries.ConfigEntry,
+    ) -> config_entries.OptionsFlow:
+        """Get the options flow for this handler."""
+        return SteamOptionsFlowHandler(config_entry)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initiated by the user."""
+        errors = {}
+        if user_input is None and self.entry:
+            user_input = {CONF_ACCOUNT: self.entry.data[CONF_ACCOUNT]}
+        elif user_input is not None:
+            try:
+                res = await self.hass.async_add_executor_job(validate_input, user_input)
+                if res[0] is not None:
+                    name = str(res[0]["personaname"])
+                else:
+                    errors = {"base": "invalid_account"}
```

Comment:
We don't need to create a new dict every time. We create an empty dict at the top of the body. We can just set the key in that dict.

```py
errors[key] = value
```